Mathlib/Algebra/Order/Star/Basic.lean

```diff
@@ -322,38 +322,29 @@ end MulOpposite
 section StarModule
 
 variable {A : Type*} [Semiring R] [PartialOrder R] [StarRing R] [StarOrderedRing R]
-  [NonUnitalRing A] [StarRing A] [PartialOrder A] [StarOrderedRing A] [Module R A]
-  [StarModule R A] [NoZeroSMulDivisors R A] [IsScalarTower R A A] [SMulCommClass R A A]
-
-lemma StarModule.smul_lt_smul_of_pos {a b : A} {c : R} (hab : a < b) (hc : 0 < c) :
-    c • a < c • b := by
-  rw [← sub_pos] at hab ⊢
-  rw [← smul_sub]
-  refine lt_of_le_of_ne ?le ?ne
-  case le =>
-    have hab := le_of_lt hab
-    rw [StarOrderedRing.nonneg_iff] at hab ⊢
-    refine AddSubmonoid.closure_induction ?mem ?zero ?add hab
-    case mem =>
-      intro x hx
-      have hc := le_of_lt hc
-      rw [StarOrderedRing.nonneg_iff] at hc
-      refine AddSubmonoid.closure_induction ?memc ?zeroc ?addc hc
-      case memc =>
-        intro c' hc'
-        obtain ⟨z, hz⟩ := hc'
-        obtain ⟨y, hy⟩ := hx
-        apply AddSubmonoid.subset_closure
-        refine ⟨z • y, ?_⟩
-        simp only [star_smul, smul_mul_smul_comm, hz, hy]
-      case zeroc => simp only [zero_smul, zero_mem]
-      case addc => exact fun c' d _ _ ↦ by simpa only [add_smul] using add_mem
-    case zero => simp only [smul_zero, zero_mem]
-    case add => exact fun x y _ _ ↦ by simpa only [smul_add] using add_mem
-  case ne =>
-    refine (smul_ne_zero ?_ ?_).symm
-    · exact (ne_of_lt hc).symm
-    · exact (ne_of_lt hab).symm
+  [NonUnitalRing A] [StarRing A] [PartialOrder A] [StarOrderedRing A] [Module R A] [StarModule R A]
+  [IsScalarTower R A A] [SMulCommClass R A A]
+
+instance : PosSMulMono R A := by
```

Comment:
You might as well just go directly for `IsStrictOrderedModule` here, no?

Reply:
This comment belongs in #28852